Layer/Pdo/Iterator.php

```diff
@@ -0,0 +1,179 @@
+ *
+ * Iterator Statement.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+class Iterator implements Database\IDal\WrapperIterator
+{
+    /**
+     * The statement instance.
+     *
+     * @var \PDOStatement
+     */
+    protected $_statement   = null;
+
+    /**
+     * The cursor orientation.
+     *
+     * @var int
+     */
+    protected $_orientation = 0;
+
+    /**
+     * The start cursor offset.
+     *
+     * @var int
+     */
+    protected $_offset      = 0;
+
+    /**
+     * The cursor row.
+     *
+     * @var array
+     */
+    protected $_row         = null;
+
+
+
+    /**
+     * Create an iterator instance.
+     *
+     * @param   \PDOStatement  $statement      The PDOStatement instance.
+     * @param   int            $orientation    This value must be one of the
+     *                                         DalStatement::FETCH_ORI_*
+     *                                         constants.
+     * @param   int            $offset         This value can be one of the
+     *                                         DalStatement::FETCH_OFFSET_*
+     *                                         constants or an arbitrary offset.
+     * @return  void
+     */
+    public function __construct(\PDOStatement $statement, $orientation, $offset)
+    {
+        $this->_statement   = $statement;
+        $this->_orientation = $orientation;
+        $this->_offset      = $offset;
+
+        return;
+    }
+
+    /**
+     * Get the statement instance.
+     *
+     * @return  \PDOStatement
+     */
+    protected function getStatement()
+    {
+        return $this->_statement;
+    }
+
+    /**
+     * Return the current element.
+     *
+     * @return  array
+     */
+    public function current()
+    {
+        return $this->_row;
+    }
+
+    /**
+     * Move forward to next element.
+     *
+     * @return  void
+     */
+    public function next()
+    {
+        $this->_row = $this->getStatement()->fetch(
+            \PDO::FETCH_ASSOC,
+            $this->_orientation
+        );
+
+        return;
+    }
+
+    /**
+     * Return always null.
+     *
+     * @return  null
+     */
+    public function key()
+    {
+        return null;
+    }
+
+    /**
+     * Checks if current position is valid.
+     *
+     * @return  bool
+     */
+    public function valid()
+    {
+        return false !== $this->_row;
+    }
+
+    /**
+     * Rewind the Iterator to the first element.
+     *
+     * @return  void
+     */
+    public function rewind()
+    {
+        if (Database\DalStatement::FETCH_OFFSET_LAST === $this->_offset) {
+            $this->_row = $this->getStatement()->fetch(
+                \PDO::FETCH_ASSOC,
+                \PDO::FETCH_ORI_LAST
+            );
+        } else {
+            $this->_row = $this->getStatement()->fetch(
+                \PDO::FETCH_ASSOC,
+                \PDO::FETCH_ORI_ABS,
+                $this->_offset
+            );
+        }
```

Comment:
If I understood correctly, `closeCursor` will allow to **reexecute** the query, and this is not what we want. However, a new iterator should be re-generate, but this is part of the user to deal with it I guess.